### PR TITLE
GraphQL: Deprecate id attributes in favor of uids

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) deprecates multiple instances of `id` and similar attributes in favor of `uid` attributes. This PR also updates numerous examples that used the deprecated attributes.

~~It'll probably take multiple PRs to complete this work.~~

The `CartItemInterface` and its implementations were not previously documented, but they were added in this PR, because it was one of the areas most affected by these changes.

This PR excludes changes to wishlists and RMA, because there are other PRs in progress. It also excludes any changes made to B2B.

Related ticket: MC-38340

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

whatsnew
In GraphQL, we've deprecated many `id` fields that were of type `Int` in favor of `uid` fields of type `ID`. These changes primarily affect the product and cart sections of the schema.

### CODE REVIEW

File: src/_includes/graphql/category-filter-input.md [hunk 1]
```diff
@@ -2,7 +2,8 @@ The `CategoryFilterInput` object defines the filters to be used in this query.
 
 Attribute | Data type | Description
 --- | --- | ---
-`ids` | FilterEqualTypeInput | Filters by the specified category IDs
+`category_uid` | FilterEqualTypeInput | Filters by the unique ID of a category for objects that implement `CategoryInterface`
+`ids` | FilterEqualTypeInput | Deprecated. Use `category_uid` insteadFilters by the specified category IDs
```
Comment (rogyar) [inline src/_includes/graphql/category-filter-input.md:6]:
```suggestion
`ids` | FilterEqualTypeInput | Deprecated. Use `category_uid` instead. Filters by the specified category IDs
```

**Review by rogyar — COMMENTED** [review]
Hi @keharper. Fantastic piece of work, thank you for covering that. Also, please take a look at the `url-resolver.md` line 64 (cannot highlight it inline because of the known GitHub issue). I believe, It's supposed to be "Deprecated. Use `entity_uid`..." instead.

**Review by erikmarr — APPROVED** [review]
Looks good. Just some minor edits.

**Review by cpartica — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**